### PR TITLE
refactor: Refactor main.py to prepare city mode in gui

### PR DESCRIPTION
**Pull request by JulesL2**
## Description

DRAFT : A lot of things will be cleaned when having a working sample

For now I have issues managing the button and preview location, the download button still work but the preview does and is misplaced.

This PR refactor main.py to prepare for the inclusion of city mode in the gui

### CODE REVIEW

**Comment by JulesL2** [comment]
I am open to any suggestion and comments @ThomasParistech

File: pretty_gpx/main.py [hunk 1]
```diff
@@ -5,7 +5,10 @@
 import tempfile
```
Comment (ThomasParistech) [inline pretty_gpx/main.py:5]: Hi @JulesL2,

Thanks for your input! I’ve actually been working on incorporating the city mode into the GUI as well. I’m using the page element of NiceGUI to create a sidebar menu that allows users to switch between the city and mountain modes, with each rendering mode having its own dedicated page.

I’m planning to open a PR tomorrow or over the weekend

Apologies for any redundancy in effort—I appreciate your contribution!

Comment (JulesL2) [reply pretty_gpx/main.py:5]: Hi @ThomasParistech

Nice update 😀 looking forward to have the PR merged 😄

Comment (ThomasParistech) [reply pretty_gpx/main.py:5]: By the way, when running the app on the trail des 4 chateaux, i got this error. Does it also raise the error when you run it (after emptying the cache)?

TypeError: Unexpected member type <class 'overpy.RelationNode'> not in [RelationWay, RelationRelation]

**Pull request closed without merging**